test/SlowTests/Issues/RavenDB-17068.cs

```diff
@@ -0,0 +1,157 @@
+
+            session.Store(o1);
+            session.Store(o2);
+
+            session.SaveChanges();
+
+            var index = new DummyIndex();
+            
+            await index.ExecuteAsync(store);
+            Indexes.WaitForIndexing(store);
+            Tuple<bool, DynamicJsonValue> alertRaised;
+
+            do
+            {
+                alertRaised = await notificationsQueue.TryDequeueAsync(TimeSpan.FromSeconds(5));
+                
+            } while (alertRaised.Item2["Type"].ToString() != NotificationType.AlertRaised.ToString());
+            
+            var details = alertRaised.Item2[nameof(AlertRaised.Details)] as DynamicJsonValue;
+            
+            using (var ctx = JsonOperationContext.ShortTermSingleUse())
+            {
+                var json = ctx.ReadObject(details, "foo");
+
+                var detailsObject = DocumentConventions.DefaultForServer.Serialization.DefaultConverter.FromBlittable<MismatchedReferencesLoadWarning>(json, "Warnings");
+                
+                Assert.Equal("DummyIndex",detailsObject.IndexName);
+                Assert.Equal(1, detailsObject.Warnings.Count);
+
+                detailsObject.Warnings.TryGetValue("orders/2-A", out var warnings);
+                
+                Assert.NotNull(warnings);
+                Assert.Equal(1, warnings.Count);
+                Assert.Equal("animals/1-A", warnings.First().ReferenceId);
+                Assert.Equal("orders/2-A", warnings.First().SourceId);
+                Assert.Equal("Animals", warnings.First().ActualCollection);
+                Assert.Equal(2, warnings.First().MismatchedCollections.Count);
+            }
+        }
+    }
+
+    [RavenFact(RavenTestCategory.Indexes)]
+    public async Task CheckIfNotificationIsNotSendWithAllLoadsEventuallyMatching()
+    {
+        using var store = GetDocumentStore();
+        var db = await GetDatabase(store.Database);
+        
+        var notificationsQueue = new AsyncQueue<DynamicJsonValue>();
+        
+        using (db.NotificationCenter.TrackActions(notificationsQueue, null))
+        using (var session = store.OpenSession())
+        {
+            var d1 = new Dog() { Name = "CoolDog1" };
+            var d2 = new Dog() { Name = "CoolDog2" };
+
+            session.Store(d1);
+            session.Store(d2);
+            
+            session.SaveChanges();
+
+            var index = new DummyIndex();
+            
+            await index.ExecuteAsync(store);
+            Indexes.WaitForIndexing(store);
+            Tuple<bool, DynamicJsonValue> alertRaised = await notificationsQueue.TryDequeueAsync(TimeSpan.FromSeconds(5));
+            
+            while (alertRaised.Item1)
+            {
+                Assert.NotEqual(NotificationType.AlertRaised.ToString(), alertRaised.Item2["Type"].ToString());
+                alertRaised = await notificationsQueue.TryDequeueAsync(TimeSpan.FromSeconds(5));
+            }
+        }
+    }
+    
```

Comment:
@Lwiel Maybe It could be nice idea to add a small unit test playing with `MismatchedReferencesWarningHandler` directly?

Reply:
Done